src/cache/index.ts

```diff
@@ -36,7 +36,8 @@ export const transaction = <T = void>(
         const handleResult = handler(tx.objectStore(storeName))
         if (handleResult) {
           ;(handleResult as IDBRequest<T>).onsuccess = event => {
-            resolve(((event.target as unknown) as { result: T }).result)
+            const { result } = (event.target as unknown) as { result: T }
+            if (result !== undefined) resolve(result)
```

Comment:
Unrelated to the rest of the PR but this fixes a bug where it would look for nonexistent items in the cache, and resolve with undefined even though the value doesn't exist in the cache